force-app/main/default/classes/DMLOperations.cls

```diff
@@ -180,16 +228,44 @@ public with sharing class DMLOperations {
 	 * @param contactsToUpsert List of Contacts to be upserted and linked to accounts.
 	 */
 	public static void upsertAccountsWithContacts(List<Contact> contactsToUpsert) {
-		// Loop through each Contact
-
+        Set<String> lastNames = new Set<String>();
+        for (Contact con : contactsToUpsert) {
+            lastNames.add(con.LastName);
+        }
```

Comment:
You can dramatically reduce the amount of code you wrote for this method by taking advantage of the previous method, `upsertAccount(String accountName)` as recommended in this method's hint. If you find the time, try to refactor the code in this method to shorten it up quite a bit.